diesel/src/mysql/connection/url.rs

```diff
@@ -59,13 +90,17 @@ impl ConnectionOptions {
             Some(segment) => Some(CString::new(segment.as_bytes())?),
         };
 
+        // this is not present in the database_url, using a default value
```

Comment:
Maybe some people have a use for the old behavior though.
Perhaps ultimately we want to make the connection open-able from `ConnectionOptions` and implement [`FromStr`](https://doc.rust-lang.org/stable/std/str/trait.FromStr.html) on it, as well as some things to update the flags that don't break the typing, such as this one?
May be slightly drifting off-topic but I feel like the requirement of [`establish`](https://docs.diesel.rs/master/diesel/prelude/trait.Connection.html#tymethod.establish) on `Connection` is a bit too much: it's not super flexible as it requires that the url is the only input, and it has to be something that always errors on r2d2 connections... It feels like it's something each connection should implement separately.

Not a blocker for me though.

Reply:
Yes, that was my impression as well, `establish()` is too restrictive, there's no way to manipulate ConnectionOptions outside of `parse()` currently. This is understandable of course, because ConnectionOptions is MySQL specific currently, so there could not be any ways to manipulate it through the Connection trait which is not database engine specific.
I'm not sure if there's a use for something like ConnectionOptions for other database engines, but if so, maybe there could be a non-db-specific ConnectionOptions trait in the future?